File: redome/src/main/java/br/org/cancer/modred/vo/LocusVO.java
package br.org.cancer.modred.vo;

import java.io.Serializable;

/**
 * Classe para passagem de parâmetro entre front e back-end para validação do HLA.
 * 
 * @author Queiroz
 */
public class LocusVO implements Serializable {
    private static final long serialVersionUID = -8536060874590895638L;

    private String codigo;
    private String primeiroAlelo;
    private String segundoAlelo;

    /**
     * Código do locus.
     * 
     * @return codigo do locus
     */
    public String getCodigo() {
        return codigo;
    }

    /**
     * Seta o codigo do locus.
     * 
     * @param codigo
     */
    public void setCodigo(String codigo) {
        this.codigo = codigo;
    }

    /**
     * Retorna o valor do primeiro alelo.
     * @return
     */
    public String getPrimeiroAlelo() {
        return primeiroAlelo;
    }

    /**
     * Seta o valor do primeiro alelo.
     * @param primeiroAlelo
     */
    public void setPrimeiroAlelo(String primeiroAlelo) {
        this.primeiroAlelo = primeiroAlelo;
    }

    /**
     * Retorna o valor do segundo alelo.
     * @return
     */
    public String getSegundoAlelo() {
        return segundoAlelo;
    }

    /**
     * Seta o valor do segundo alelo.
     * @param segundoAlelo
     */
    public void setSegundoAlelo(String segundoAlelo) {
        this.segundoAlelo = segundoAlelo;
    }
}

File: redome-front/src/app/paciente/cadastro/exame/locusvo.ts
import { BaseEntidade } from "../../../shared/base.entidade";

/**
 * Classe domínio de Locus
 * @author Fillipe Queiroz
 * @export
 * @class Locus
 * @extends {BaseEntidade}
 */
export class LocusVO extends BaseEntidade {

  public static LOCI_A : string = "A";
  public static LOCI_B : string = "B";
  public static LOCI_C : string = "C";  
  public static LOCI_DRB1 : string = "DRB1";
  public static LOCI_DQB1 : string = "DQB1";

    private _codigo: String;
    private _primeiroAlelo: String;
    private _segundoAlelo: String;

	/**
     * 
     * 
     * @type {String}
     * @memberof LocusVO
     */
    public get codigo(): String {
      return this._codigo;
    }

	/**
     * 
     * 
     * 
     * @memberof LocusVO
     */
    public set codigo(value: String) {
      this._codigo = value;
    }

	/**
     * Retorna o valor do primeiro alelo.
     * 
     * @type {String}
     * @memberof LocusVO
     */
    public get primeiroAlelo(): String {
      return this._primeiroAlelo;
    }

	/**
     * Seta o valor do primeiro alelo.
     * 
     * @memberof LocusVO
     */
    public set primeiroAlelo(value: String) {
      this._primeiroAlelo = value;
    }

  	/**
     * Retorna o valor do segundo alelo.
     * 
     * @type {String}
     * @memberof LocusVO
     */
    public get segundoAlelo(): String {
      return this._segundoAlelo;
    }

	/**
     * Seta o valor do segundo alelo.
     * 
     * @memberof LocusVO
     */
    public set segundoAlelo(value: String) {
      this._segundoAlelo = value;
    }

    public jsonToEntity(res: any) {
      throw new Error("Method not implemented.");
  }

}